src/components/ProjectPreview.jsx

```diff
@@ -1,14 +1,50 @@
+import classnames from 'classnames';
 import React from 'react';
 import PropTypes from 'prop-types';
 import moment from 'moment';
 
 const MAX_LENGTH = 50;
 
-export default function ProjectPreview({preview, project}) {
+export default function ProjectPreview({
+  preview,
+  project,
+  onProjectArchived,
+  onProjectUnArchived,
+}) {
   return (
     <div>
-      <div className="project-preview__label">
-        {preview.slice(0, MAX_LENGTH)}
+      <div className="project-preview">
+        <div
+          className={classnames('project-preview__label', {
+            'project-preview__label_archived': project.archived,
+          })}
+        >
+          {preview.slice(0, MAX_LENGTH)}
+        </div>
+        {project.archived ?
+          (
+            <div
+              className="project-preview__archive"
+              onClick={(e) => {
+                e.stopPropagation();
+                onProjectUnArchived();
+              }}
+            >
+              <span className="u__icon">&#xf0aa;</span>
+            </div>
+          ) :
+          (
+            <div
+              className="project-preview__archive"
+              onClick={(e) => {
+                e.stopPropagation();
+                onProjectArchived();
```

Comment:
Probably not worth worrying about in this PR, but perhaps for a followup—the disappearance of the project from the list when you click the unarchive button feels a bit abrupt—would be nice to get a fade and/or slide animation popping. (This also might fall in the set of things that are way more annoying in a strict Redux-based state architecture, although I would be surprised if there weren’t some good libraries for abstracting this away at the React level)

Reply:
Will look into this!